\begin{abstract}
Whole-genome sequencing is increasingly used to identify Mendelian variants in clinical pipelines. These pipelines focus on single-nucleotide variants (SNVs) and also structural variants, while ignoring more complex repeat sequence variants.
For example, Variable Number Tandem Repeats (VNTRs), composed of inexact tandem duplications of short (6-100 bp) repeating units have been implicated in multiple Mendelian disorders. However, genotyping these regions and identifying their structural from sequencing data has been mostly ignored due to computational difficulties.
One of the computationally expensive parts of genotyping these variations from NGS data is identifying the reads that contains these variants.
We propose a novel method for classification of the sequencing reads using neural networks for targeted alignment of the reads to specific locations of the genome. Through our experiments, we show that it provides higher accuracy than the existing sensitive alignment tools while its running time is comparable to the fast general purpose aligners.
\paragraph{Keywords.} Tandem Repeat, eQTL, Gene Regulation
\end{abstract}
